codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/EventStreamTestTools.kt

```diff
@@ -0,0 +1,174 @@
+        val generator = requirements.renderGenerator(codegenContext, test, protocol)
+
+        test.project.lib {
+            when (variety) {
+                EventStreamTestVariety.Marshall -> writeMarshallTestCases(testCase, generator)
+                EventStreamTestVariety.Unmarshall -> writeUnmarshallTestCases(testCase, codegenTarget, generator)
+            }
+        }
+        test.project.compileAndTest()
+    }
+
+    private fun <C : CodegenContext> generateTestProject(
+        requirements: EventStreamTestRequirements<C>,
+        codegenContext: C,
+        codegenTarget: CodegenTarget,
+    ): TestEventStreamProject {
+        val model = codegenContext.model
+        val symbolProvider = codegenContext.symbolProvider
+        val operationShape = model.expectShape(ShapeId.from("test#TestStreamOp")) as OperationShape
+        val unionShape = model.expectShape(ShapeId.from("test#TestStream")) as UnionShape
+
+        val project = TestWorkspace.testProject(symbolProvider)
+        val operationSymbol = symbolProvider.toSymbol(operationShape)
+        project.withModule(ErrorsModule) {
+            val errors = model.shapes()
+                .filter { shape -> shape.isStructureShape && shape.hasTrait<ErrorTrait>() }
+                .map { it.asStructureShape().get() }
+                .toList()
+            when (codegenTarget) {
+                CodegenTarget.CLIENT -> CombinedErrorGenerator(model, symbolProvider, operationSymbol, errors).render(this)
+                CodegenTarget.SERVER -> ServerCombinedErrorGenerator(model, symbolProvider, operationSymbol, errors).render(this)
+            }
+            for (shape in model.shapes().filter { shape -> shape is StructureShape && shape.hasTrait<ErrorTrait>() }) {
+                StructureGenerator(model, symbolProvider, this, shape as StructureShape).render(codegenTarget)
+                requirements.renderBuilderForShape(this, codegenContext, shape)
+            }
+        }
+        project.withModule(ModelsModule) {
+            val inputOutput = model.lookup<StructureShape>("test#TestStreamInputOutput")
+            recursivelyGenerateModels(model, symbolProvider, inputOutput, this, codegenTarget)
+        }
+        project.withModule(RustModule.Output) {
+            operationShape.outputShape(model).renderWithModelBuilder(model, symbolProvider, this)
+        }
+        return TestEventStreamProject(
+            model,
+            codegenContext.serviceShape,
+            operationShape,
+            unionShape,
+            symbolProvider,
+            project,
+        )
+    }
+
+    private fun recursivelyGenerateModels(
+        model: Model,
+        symbolProvider: RustSymbolProvider,
+        shape: Shape,
+        writer: RustWriter,
+        mode: CodegenTarget,
+    ) {
+        for (member in shape.members()) {
+            if (member.target.namespace == "smithy.api") {
+                continue
+            }
```

Comment:
Why do we need to check the namespace like this. What happens if we don't?

Reply:
This filters out the Smithy prelude types, such as `smithy.api#Blob` or `smithy.api#String`.

Reply:
Ah, ok